commit bd730951069ec168cbbe4fb490b0d40069e4408a (refs/changes/90/390/2)
Author: Pedro P. Candel <pedro@joyent.com>
Date:   2016-08-31T08:07:32+02:00 (3 years, 1 month ago)
    
    TOOLS-1538 sdcadm should not throw ENOENT when checking for sdc-usbkey

diff --git a/lib/common.js b/lib/common.js
index a565ceb..5864508 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -1295,12 +1295,17 @@ function isUsbKeyMounted(log, cb) {
     function sdcUsbkeyExists(ctx, next) {
         fs.stat(ctx.utility, function (er, st) {
             if (er) {
-                return next(er);
+                if (er.code === 'ENOENT') {
+                    next();
+                } else {
+                    next(er);
+                }
+                return;
             }
             if (st.isFile()) {
                 ctx.utilityExists = true;
             }
-            return next();
+            next();
         });
     }
 
@@ -1330,11 +1335,17 @@ function isUsbKeyMounted(log, cb) {
 
     function checkMeyMounted(ctx, next) {
         if (ctx.keyIsMounted || ctx.utilityExists) {
-            return next();
+            next();
+            return;
         }
         fs.stat(ctx.key, function (er, st) {
             if (er) {
-                return next(er);
+                if (er.code === 'ENOENT') {
+                    next();
+                } else {
+                    next(er);
+                }
+                return;
             }
             if (st.isFile()) {
                 ctx.keyIsMounted = true;
@@ -1382,7 +1393,12 @@ function mountUsbKey(log, cb) {
         }
         fs.stat(ctx.utility, function (er, st) {
             if (er) {
-                return next(er);
+                if (er.code === 'ENOENT') {
+                    next();
+                } else {
+                    next(er);
+                }
+                return;
             }
             if (st.isFile()) {
                 ctx.utilityExists = true;
@@ -1481,16 +1497,23 @@ function unmountUsbKey(log, cb) {
 
     function sdcUsbkeyExists(ctx, next) {
         if (!ctx.keyIsMounted) {
-            return next();
+            next();
+            return;
         }
         fs.stat(ctx.utility, function (er, st) {
             if (er) {
-                return next(er);
+                if (er.code === 'ENOENT') {
+                    next();
+                } else {
+                    next(er);
+                }
+                return;
             }
             if (st.isFile()) {
                 ctx.utilityExists = true;
             }
-            return next();
+            next();
+            return;
         });
     }
 
